rust/src/delta.rs

```diff
@@ -1349,32 +1358,51 @@ impl<'a> DeltaTransaction<'a> {
     /// This method will retry the transaction commit based on the value of `max_retry_commit_attempts` set in `DeltaTransactionOptions`.
     pub async fn commit(
         &mut self,
-        operation: Option<DeltaOperation>,
+        operation: DeltaOperation,
         app_metadata: Option<Map<String, Value>>,
     ) -> Result<DeltaDataTypeVersion, DeltaTableError> {
-        // TODO: stubbing `operation` parameter (which will be necessary for writing the CommitInfo action),
-        // but leaving it unused for now. `CommitInfo` is a fairly dynamic data structure so we should work
-        // out the data structure approach separately.
+        // TODO(roeap) in the reference implementation this logic is implemented, which seems somewhat strange,
```

Comment:
In the master the default isolation level seems to be retrieved from table metadata: https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala#L1213